### PR TITLE
fix: #3073

### PR DESCRIPTION
**Pull request by HHongSeungWoo**
I have fixed the bug.

before

https://github.com/recharts/recharts/assets/34399997/6cae6da9-d761-4864-a3a5-442b0e6d3fa5

after

https://github.com/recharts/recharts/assets/34399997/6156aa9c-eeaf-4b95-bbe0-bd77c8c5dbde

## Related Issue

#3073

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by ckifer** [comment]
@HHongSeungWoo do you know why this worked on some browsers and not others?

**Comment by HHongSeungWoo** [comment]
@ckifer If you are referring to Firefox DevTools, Firefox's mobile mode does not simulate mobile devices like Chrome does. It shows mobile pages but triggers mouse events.

**Comment by ckifer** [comment]
Oh wow that's good to know 🙃. 

 Cool, will review this more later

File: src/chart/generateCategoricalChart.tsx [hunk 1]
```diff
@@ -1116,8 +1114,7 @@ export const generateCategoricalChart = ({
     constructor(props: CategoricalChartProps) {
       super(props);
 
-      this.uniqueChartId = _.isNil(props.id) ? uniqueId('recharts') : props.id;
-      this.clipPathId = `${this.uniqueChartId}-clip`;
+      this.clipPathId = `${props.id ?? uniqueId('recharts')}-clip`;
```
Comment (nikolasrieble) [inline src/chart/generateCategoricalChart.tsx:1117]: Nice. Love the small cleanups.

File: src/chart/generateCategoricalChart.tsx [hunk 2]
```diff
@@ -1423,21 +1420,12 @@ export const generateCategoricalChart = ({
...
-        eventCenter.setMaxListeners(eventCenter._maxListeners + 1);
-      }
     }
 
     removeListener() {
       eventCenter.removeListener(SYNC_EVENT, this.handleReceiveSyncEvent);
-
-      if (eventCenter.setMaxListeners && eventCenter._maxListeners) {
-        eventCenter.setMaxListeners(eventCenter._maxListeners - 1);
-      }
```
Comment (nikolasrieble) [inline src/chart/generateCategoricalChart.tsx:1440]: Could you provide some context as to why we can remove this?

File: src/chart/generateCategoricalChart.tsx [hunk 3]
```diff
@@ -1659,11 +1645,7 @@ export const generateCategoricalChart = ({
     };
 
     triggerSyncEvent(data: CategoricalChartState) {
-      const { syncId } = this.props;
-
-      if (!_.isNil(syncId)) {
-        eventCenter.emit(SYNC_EVENT, syncId, this.uniqueChartId, data);
-      }
```
Comment (nikolasrieble) [inline src/chart/generateCategoricalChart.tsx:1666]: Same here, why can we remove this? Can the syncId no longer be null?

Comment (nikolasrieble) [reply src/chart/generateCategoricalChart.tsx:1666]: How are we making sure it is not null? Could a user not still set it to null?

File: test/chart/LineChart.spec.tsx [hunk 4]
```diff
@@ -751,7 +750,7 @@ describe('<LineChart /> - Rendering two line charts with syncId', () => {
     expect(container.querySelectorAll('.recharts-tooltip-cursor')).toHaveLength(2);
 
     // make sure tooltips display the correct values, synced by data value
-    expect(screen.queryByText('400')).toBeTruthy();
+    expect(screen.queryByText('300')).toBeTruthy();
```
Comment (nikolasrieble) [inline test/chart/LineChart.spec.tsx:753]: This is surprising. Did the test previously capture behaviour that was a bug itself? 
Could you please add a storybook test (i.e. in a play function for a story with syncId) to verify that the syncId still works correctly?

**Review by nikolasrieble — COMMENTED** [review]
Looks like a cleanup mostly, I have a few questions. Thank you for your contribution.

File: src/chart/generateCategoricalChart.tsx (hunk 2 above)
Comment (ckifer) [reply src/chart/generateCategoricalChart.tsx:1440]: +1

File: src/chart/generateCategoricalChart.tsx (hunk 3 above)
Comment (ckifer) [reply src/chart/generateCategoricalChart.tsx:1666]: +1

File: src/chart/generateCategoricalChart.tsx (hunk 2 above)
Comment (HHongSeungWoo) [reply src/chart/generateCategoricalChart.tsx:1440]: `setMaxListeners` is not a valid method.

File: src/chart/generateCategoricalChart.tsx (hunk 3 above)
Comment (HHongSeungWoo) [reply src/chart/generateCategoricalChart.tsx:1666]: It was just a missed. Thank you for finding

File: test/chart/LineChart.spec.tsx (hunk 4 above)
Comment (HHongSeungWoo) [reply test/chart/LineChart.spec.tsx:753]: Great, I'll give it a try.

File: src/chart/generateCategoricalChart.tsx (hunk 2 above)
Comment (ckifer) [reply src/chart/generateCategoricalChart.tsx:1440]: https://github.com/primus/eventemitter3/commit/315f5a0ea0b00005b1ff584a9456071e14f7bd8e
proof in the pudding, nice

File: src/chart/generateCategoricalChart.tsx [hunk 5]
```diff
@@ -1659,10 +1645,8 @@ export const generateCategoricalChart = ({
     };
 
     triggerSyncEvent(data: CategoricalChartState) {
-      const { syncId } = this.props;
-
-      if (!_.isNil(syncId)) {
-        eventCenter.emit(SYNC_EVENT, syncId, this.uniqueChartId, data);
+      if (this.props.syncId !== undefined) {
```
Comment (nikolasrieble) [inline src/chart/generateCategoricalChart.tsx:1648]:
```suggestion
      if (this.props.syncId != undefined) {
```
I assume we would also want to catch null, right?

Comment (HHongSeungWoo) [reply src/chart/generateCategoricalChart.tsx:1648]: No, there shouldn't be any null. Typing errors should be caught by TypeScript.

**Review by ckifer — APPROVED** [review]
I'm satisfied with this, I think since we've just done a release there is plenty of time to see if this causes any regressions. Not too worried about that though

**Pull request merged**